.github/workflows/check-pr-engine-editor-debug-and-tests.yaml

```diff
@@ -0,0 +1,90 @@
+name: Check PR - Engine Editor Debug And Tests
+on:
+  pull_request:
+    paths:
+      - 'src/**'
+      - 'config.py'
+      - 'register_types.cpp'
+      - 'register_types.h'
+      - 'SCsub'
+      - 'kt/**'
+      - 'harness/tests/**'
+      - '.github/workflows/check-pr-engine-editor-debug-and-tests.yaml'
+
+jobs:
+  build:
+    strategy:
+      matrix:
+        os: [ubuntu-latest, macos-latest, windows-latest]
+        include:
+          - os: ubuntu-latest
+            platform: x11
+          - os: macos-latest
+            platform: osx
+          - os: windows-latest
+            platform: windows
+    runs-on: ${{ matrix.os }}
+    steps:
+      - name: Configure dependencies
+        if: matrix.platform == 'x11'
+        run: |
+          sudo apt-get install build-essential pkg-config libx11-dev libxcursor-dev \
+            libxinerama-dev libgl1-mesa-dev libglu-dev libasound2-dev libpulse-dev libudev-dev libxi-dev libxrandr-dev yasm
+      - name: Clone Godot Engine
+        uses: actions/checkout@v2
+        with:
+          repository: godotengine/godot
+          ref: 3.2.3-stable
+      - name: Clone Godot JVM module.
+        uses: actions/checkout@v2
+        with:
+          path: modules/kotlin_jvm
+          submodules: recursive
+      - name: Set up Python 3.x
+        uses: actions/setup-python@v2
+        with:
+          python-version: '3.x'
+          architecture: 'x64'
+      - name: Configuring Python packages
+        run: |
+          python -c "import sys; print(sys.version)"
+          python -m pip install scons
+      - name: Build with editor debug
+        run: |
+          scons platform=${{ matrix.platform }}
+      - name: Upload OSX binary
+        if: matrix.platform == 'osx'
+        uses: actions/upload-artifact@v1
+        with:
+          name: osx-editor-debug-app
+          path: bin/godot.osx.tools.64
+  test:
+    needs: build
+    runs-on: macos-latest

```

Comment:
Why do we run the tests only on osx?

Reply:
IIRC gut cannot run from command line on windows. I do not remember for Windows. I can try it.

Reply:
IIRC Linux cannot run because we have no graphics on CI

Reply:
Hmm we could use a headless build i guess. But yeah overkill for now and something to maybe lookt at at a later stage